docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -2029,6 +2029,148 @@ transformations themselves. You can use Daml Script (with Canton) to test some
 isolated simple cases, but for thorough tests of you system using SCU, you should
 prefer full workflow testing, as below.
 
+Multi-package builds for upgrades
+--------------------------------------
+.. _multi_package_upgrades:
+
+When developing upgrades, you will likely have multiple DARs in scope that need
+to be built together - tracking these DARs and building them in the right order
+can be complicated, especially as you develop live and as the project grows.
+
+:ref:`Multi-package builds <multi-package-build>` are a new feature that has exited beta in 2.10 which helps
+with projects containing multiple DARs, for example a project using upgrades.
+
+To get started understanding how the multi-package builds simplify the
+development of a project using upgrades, begin by creating a new Daml project
+with the ``upgrades-example`` template.
+
+.. code:: bash
+
+   > daml new upgraded-iou --template upgrades-example
+   > cd upgraded-iou
+   > tree
+   .
+   ├── multi-package.yaml
+   ├── run-test.sh
+   ├── interfaces
+   │   ├── daml/...
+   │   └── daml.yaml
+   ├── main-v1
+   │   ├── daml/...
+   │   └── daml.yaml
+   ├── main-v2
+   │   ├── daml/...
+   │   └── daml.yaml
+   └── test
+       ├── daml/...
+       └── daml.yaml
+
+The example template contains:
+
+- A package ``upgraded-iou-interfaces``, which defines an interface ``Asset``
+  and a viewtype ``Asset.View``.
+- The first version (1.0.0) of a package ``upgraded-iou-main``, which defines a
+  template ``IOU`` with instance of ``upgraded-iou-interfaces:Main.Asset``.
+- The second version (2.0.0) of ``upgraded-iou-main`` which upgrades
+  the first. It adds a new ``description`` field to ``IOU``, and uses it (when
+  the field is defined) in an upgraded implementation of ``Asset``.
+- A testing package ``upgraded-iou-test``, which depends on both
+  ``upgraded-iou-main-1.0.0`` and ``upgraded-iou-main-2.0.0``. It defines a
+  script which exercises v1.0.0 and v2.0.0 ``IOU``s via their ``Asset``
+  interface.
+- A script ``run-test.sh``, which runs the main test in ``upgraded-iou-test``.
+- A ``multi-package.yaml`` file which lists our four packages.
+
+Normally, without multi-package builds, you would test your program the following way:
+
+.. code:: bash
+
+   > # Run sandbox in the background, wait until the three lines below are shown
+   > daml sandbox &
+   Starting Canton sandbox.
+   Listening at port 6865
+   Canton sandbox is ready.
+   > # Build all, run test
+   > cd interfaces/; daml build --enable-multi-package=no
```

Comment:
Unsure about this one, as you wouldn't need the flag if you didn't have the multi-package.yaml.

Reply:
Because the template already contains a multi-package.yaml, I didn't want to get them to remove the multi yaml and then add it back. I also prefer the flag because it makes it obvious that multi build is being disabled for this case.